src/main/java/de/tum/in/www1/artemis/service/hestia/ExerciseHintService.java

```diff
@@ -0,0 +1,267 @@
+import de.tum.in.www1.artemis.domain.hestia.CodeHint;
+import de.tum.in.www1.artemis.domain.hestia.ExerciseHint;
+import de.tum.in.www1.artemis.domain.hestia.ExerciseHintActivation;
+import de.tum.in.www1.artemis.domain.hestia.ProgrammingExerciseTask;
+import de.tum.in.www1.artemis.repository.StudentParticipationRepository;
+import de.tum.in.www1.artemis.repository.hestia.ExerciseHintActivationRepository;
+import de.tum.in.www1.artemis.repository.hestia.ExerciseHintRepository;
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+import de.tum.in.www1.artemis.web.rest.errors.InternalServerErrorException;
+
+@Service
+public class ExerciseHintService {
+
+    private final AuthorizationCheckService authCheckService;
+
+    private final ExerciseHintRepository exerciseHintRepository;
+
+    private final ProgrammingExerciseTaskService programmingExerciseTaskService;
+
+    private final StudentParticipationRepository studentParticipationRepository;
+
+    private final ExerciseHintActivationRepository exerciseHintActivationRepository;
+
+    public ExerciseHintService(AuthorizationCheckService authCheckService, ExerciseHintRepository exerciseHintRepository,
+            ProgrammingExerciseTaskService programmingExerciseTaskService, StudentParticipationRepository studentParticipationRepository,
+            ExerciseHintActivationRepository exerciseHintActivationRepository) {
+        this.authCheckService = authCheckService;
+        this.exerciseHintRepository = exerciseHintRepository;
+        this.programmingExerciseTaskService = programmingExerciseTaskService;
+        this.studentParticipationRepository = studentParticipationRepository;
+        this.exerciseHintActivationRepository = exerciseHintActivationRepository;
+    }
+
+    /**
+     * Copies the hints of an exercise to a new target exercise by cloning the hint objects and saving them
+     * resulting in new IDs for the copied hints. The contents stay the same. On top of that, all hints in the
+     * problem statement of the target exercise get replaced by the new IDs.
+     *
+     * @param template The template exercise containing the hints that should be copied
+     * @param target   The new target exercise, to which all hints should get copied to.
+     * @return A map with the old hint id as a key and the new hint id as a value
+     */
+    public Map<Long, Long> copyExerciseHints(final ProgrammingExercise template, final ProgrammingExercise target) {
+        final Map<Long, Long> hintIdMapping = new HashMap<>();
+        target.setExerciseHints(template.getExerciseHints().stream().map(hint -> {
+            ExerciseHint copiedHint;
+            if (hint instanceof CodeHint) {
+                copiedHint = new CodeHint();
+            }
+            else {
+                copiedHint = new ExerciseHint();
+            }
```

Comment:
One idea: You could also add a method getCopy in ExerciseHint and override it in CodeHint to return the child class.